src/coreclr/tools/Common/TypeSystem/IL/CoreRTILProvider.cs

```diff
@@ -114,6 +114,28 @@ private MethodIL TryGetPerInstantiationIntrinsicMethodIL(MethodDesc method)
 
             switch (owningType.Name)
             {
+                case "Activator":
+                    {
+                        TypeSystemContext context = owningType.Context;
+                        if (methodName == "CreateInstance" && method.Signature.Length == 0 && method.HasInstantiation
+                            && method.Instantiation[0] is TypeDesc activatedType
+                            && activatedType != context.UniversalCanonType
+                            && activatedType.IsValueType
+                            && activatedType.GetParameterlessConstructor() == null)
```

Comment:
The R2R version of this method calls `GetDefaultConstructor`. Does it have a bug?

Reply:
Looks like it. dotnet/runtime#50228